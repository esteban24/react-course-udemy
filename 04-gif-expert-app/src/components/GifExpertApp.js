import React, { useState } from 'react';
import AddCategory from './AddCategory';

const GifExpertApp = () => {
    const [ categories, setCategories ] = useState(['One Punch Man', 'Samuray X', 'Dragon Ball'])

    // const handleAdd = () => {
    //     setCategories( categories => [...categories, 'HunterXHunter'] );
    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr/>

            <ol>
                {
                    categories.map( category => {
                        return <li key={ category } >{ category }</li>
                    })
                }
            </ol>
        </>
    );
};

export default GifExpertApp;