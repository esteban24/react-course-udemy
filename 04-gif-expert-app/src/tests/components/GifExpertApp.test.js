import React from "react";
import { shallow } from "enzyme"

import GifExpertApp from '../../components/GifExpertApp';

describe('GifExpertApp.js test suite', () => {
    test('should match snapshot', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show a list of categories', () => {
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);

        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
})
