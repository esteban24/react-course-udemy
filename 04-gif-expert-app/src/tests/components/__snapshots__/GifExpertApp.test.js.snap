// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp.js test suite should match snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
